src/modules/csm/csm.py

```diff
@@ -255,8 +211,69 @@ def collect_data(self, blockstamp: BlockStamp) -> bool:
 
         return self.state.is_fulfilled
 
+    def calculate_distribution(self, blockstamp: BlockStamp) -> tuple[int, dict[NodeOperatorId, int]]:
+        """Computes distribution of fee shares at the given timestamp"""
+
+        assert self.state
+
+        threshold = self.state.avg_perf - self.w3.csm.oracle.perf_leeway(blockstamp.block_hash)
+        operators_to_validators = self.module_validators_by_node_operators(blockstamp)
+
+        # Build the map of the current distribution operators.
+        distribution: dict[NodeOperatorId, int] = {}
+        for (_, no_id), validators in operators_to_validators.items():
+            if no_id in self.stuck_operators(blockstamp):
+                continue
+
+            no_share = 0
+
+            for v in validators:
+                try:
+                    aggr = self.state.data[ValidatorIndex(int(v.index))]
+                except KeyError:
+                    # It's possible that the validator is not assigned to any duty, hence it's performance
+                    # is not presented in the aggregates (e.g. exited, pending for activation etc).
+                    continue
+                else:
```

Comment:
`else` doesn't make sense here, as I see

Reply:
It's done to clearly separate the case we catch and the rest.

Reply:
there is `continue`, so it won't reach the code below